core/src/main/java/com/yahoo/imapnio/async/data/ExtendedModifiedSinceTerm.java

```diff
@@ -0,0 +1,127 @@
+package com.yahoo.imapnio.async.data;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.mail.Flags;
+import javax.mail.Message;
+import javax.mail.search.SearchTerm;
+
+import com.yahoo.imapnio.async.exception.ImapAsyncClientException;
+import com.yahoo.imapnio.async.request.EntryTypeRequest;
+
+/**
+ * This class models search-modsequence with {@code EntryTypeReq} from RFC7162. ABNF from RFC7162 is documented below.
+ *
+ * <pre>
+ * {@code
+ * 3.1.5.  MODSEQ Search Criterion in SEARCH
+ *
+ *    The MODSEQ criterion for the SEARCH (or UID SEARCH) command allows a
+ *    client to search for the metadata items that were modified since a
+ *    specified moment.
+ *    Syntax: MODSEQ [<entry-name> <entry-type-req>] <mod-sequence-valzer>
+ * ============================================================================
+ * ABNF:
+ * search-modsequence  = "MODSEQ" [search-modseq-ext] SP mod-sequence-valzer
+ *
+ * entry-name          = entry-flag-name
+ *
+ * entry-flag-name     = DQUOTE "/flags/" attr-flag DQUOTE
+ *                       ;; Each system or user-defined flag <flag>
+ *                       ;; is mapped to "/flags/<flag>".
+ *                       ;;
+ *                       ;; <entry-flag-name> follows the escape rules
+ *                       ;; used by "quoted" string as described in
+ *                       ;; Section 4.3 of [RFC3501]; e.g., for the
+ *                       ;; flag \Seen, the corresponding <entry-name>
+ *                       ;; is "/flags/\\seen", and for the flag
+ *                       ;; $MDNSent, the corresponding <entry-name>
+ *                       ;; is "/flags/$mdnsent".
+ *
+ * entry-type-resp     = "priv" / "shared"
+ *                       ;; Metadata item type.
+ *
+ * entry-type-req      = entry-type-resp / "all"
+ *                       ;; Perform SEARCH operation on a private
+ *                       ;; metadata item, shared metadata item,
+ *                       ;; or both.
+ *
+ * }
+ * </pre>
+ */
+public final class ExtendedModifiedSinceTerm extends SearchTerm {
+
+    /** The name of the metadata item. */
+    private final Flags entryName;
```

Comment:
Do we know any servers that support modified since for specific flags?

Reply:
I have tried the following example from RFC 4551 and it works for yahoo mail and gmail:
   Example 15:
      C: a SEARCH MODSEQ "/flags/\\draft" all 620162338

Reply:
I have tried the following example from RFC 4551 and it works for yahoo mail and gmail:
Example 15:
C: a SEARCH MODSEQ "/flags/\draft" all 620162338

Reply:
@yuanxinwang , Are you sure? Yahoo/Aol does not even support CONDSTORE. Supporting individual flag modseq would be quite hard even for Gmail. Let me know if you want to do a hangout and show me what you tried.